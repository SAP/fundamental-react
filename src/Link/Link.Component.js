import { Link } from '..';
import path from 'path';
import React from 'react';
import { ComponentPage, Example } from '../_playground';

export const LinkComponent = () => {
    return (
        <ComponentPage
            description='Use an **Link** component to display a link.'
            sourceModulePath={path.join(__dirname, './Link')}
            title='Link'>

            <Example
                centered
                title='States'>
                <React.Fragment>
                    <Link href='#'>Default Link</Link>
                    <Link disabled href='#'>Disabled Link</Link>
                </React.Fragment>
            </Example>
        </ComponentPage>
    );
};
