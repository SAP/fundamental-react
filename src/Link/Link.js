import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/link.css';

const classnames = classnamesBind.bind(styles);

/** Use an **Link** component to display a link. */
const Link = React.forwardRef(({ className, cssNamespace, children, disabled, subtle, ...props }, ref) => {

    const imageClasses = classnames(
        `${cssNamespace}-link`,
        {
            'is-disabled': !!disabled,
            [`${cssNamespace}-link--subtle`]: subtle
        },
        className
    );
    return (
        <a
            {...props}
            aria-disabled={disabled}
            className={imageClasses}
            ref={ref}>
            {children}
        </a>
    );
});

Link.displayName = 'Link';

Link.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Set to **true** to mark component as disabled and make it non-interactive */
    disabled: PropTypes.bool,
    /** Set to **true** to apply subtle styling */
    subtle: PropTypes.bool
};

export default withStyles(Link);
