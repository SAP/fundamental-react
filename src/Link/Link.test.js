import Link from './Link';
import { mount } from 'enzyme';
import React from 'react';

describe('<Link />', () => {
    describe('Prop spreading', () => {
        test('should allow props to be spread to the Link component', () => {
            const element = mount(
                <Link
                    data-sample='Sample'
                    href='#' />
            );

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });

    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <Link ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('A');
        expect(ref.current.className).toEqual('fd-link');
    });
});
