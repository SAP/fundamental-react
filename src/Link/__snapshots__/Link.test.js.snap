// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> create link 1`] = `
<a
  className="fd-link blue"
  href="#"
>
  Foo
</a>
`;

exports[`<Link /> create link 2`] = `
<a
  aria-disabled={true}
  className="fd-link is-disabled"
  href="#"
>
  Disabled Link
</a>
`;
